src/spatialdata/_core/query/relational_query.py

```diff
@@ -129,6 +147,255 @@ def _filter_table_by_elements(
     return table
 
 
+def _create_element_dict(
+    element_type: str, name: str, element: SpatialElement | AnnData, elements_dict: dict[str, dict[str, Any]]
+) -> dict[str, dict[str, Any]]:
+    if element_type not in elements_dict:
+        elements_dict[element_type] = {}
+    elements_dict[element_type][name] = element
+    return elements_dict
+
+
+def _right_exclusive_join_spatialelement_table(
+    element_dict: dict[str, dict[str, Any]], table: AnnData
+) -> tuple[dict[str, Any], AnnData | None]:
+    regions, region_column_name, instance_key = get_table_keys(table)
+    groups_df = table.obs.groupby(by=region_column_name)
+    mask = []
+    for element_type, name_element in element_dict.items():
+        for name, element in name_element.items():
+            if name in regions:
+                group_df = groups_df.get_group(name)
+                table_instance_key_column = group_df[instance_key]
+                if element_type in ["points", "shapes"]:
+                    element_indices = element.index
+                    element_dict[element_type][name] = None
```

Comment:
why don't you set the element to `None` for labels? I'd move this line right after the `for name, element in ...`

Reply:
changed